Havit.Blazor.Components.Web.Bootstrap/TreeViews/Internal/HxTreeViewItemInternal.razor

```diff
@@ -42,19 +41,28 @@
     }
 </div>
 
-@if (this.IsExpanded.Value && hasChildren)
+@if (hasChildren)
 {
-    foreach (var item in children)
-    {
-        <HxTreeViewItemInternal TItem="TItem"
-                        Item="item"
-                        TitleSelector="this.TitleSelector"
-                        IconSelector="this.IconSelector"
-                        CssClass="@this.CssClass"
-                        CssClassSelector="this.CssClassSelector"
-                        ChildrenSelector="this.ChildrenSelector"
-                        Level="@(this.Level + 1)"
-                        ContentTemplate="this.ContentTemplate"
-                        OnItemSelected="this.OnItemSelected" />
-    }
-}
\ No newline at end of file
+    <HxCollapse
+        @ref="collapseComponent"
+        CollapseDirection="CollapseDirection.Vertical"
+        OnShown="HandleExpansionCompleted"
+        OnHidden="HandleCollapseCompleted"
+        InitiallyExpanded="initiallyExpanded">
+
+        @foreach (var item in children)
+        {
+            <HxTreeViewItemInternal TItem="TItem"
+                            Item="item"
+                            TitleSelector="this.TitleSelector"
+                            IconSelector="this.IconSelector"
+                            CssClass="@this.CssClass"
+                            CssClassSelector="this.CssClassSelector"
+                            ChildrenSelector="this.ChildrenSelector"
+                            Level="@(this.Level + 1)"
+                            ContentTemplate="this.ContentTemplate"
+                            OnItemSelected="this.OnItemSelected" />
+        }
```

Comment:
We should probably add `@key="item"` to the `HxTreeViewItemInternal` usages (incl. the top-level iteration) to guarantee the component won't be reused if the `item` changes (can bring new result of `InitiallyExpandedSelector` etc.)